circus-train-gcp/src/main/java/com/hotels/bdp/circustrain/gcp/DistributedFileSystemPathProvider.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.gcp;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import org.apache.hadoop.fs.Path;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.hotels.bdp.circustrain.gcp.context.GCPSecurity;
+
+@Component
+public class DistributedFileSystemPathProvider {
+
+  static final String DEFAULT_HDFS_PREFIX = "hdfs:/tmp/ct-gcp-";
```

Comment:
Sorry I should have noticed this earlier but instead of `tmp` you should probable use the value of `hive.exec.scratchdir` from the HiveConf. See https://cwiki.apache.org/confluence/display/Hive/AdminManual+Configuration#AdminManualConfiguration-TemporaryFolders